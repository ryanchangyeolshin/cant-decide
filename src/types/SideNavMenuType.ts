type SideNavMenuType = {
  sideMenu: boolean,
  toggleDrawer: (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => void,
  handleOpenModal: (event: React.MouseEvent<HTMLElement>, modalType: string) => void,
}

export default SideNavMenuType;