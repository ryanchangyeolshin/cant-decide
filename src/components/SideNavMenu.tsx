import React from 'react';
import { Drawer, List, Divider, Typography, ListItem, ListItemIcon, ListItemText, makeStyles, createStyles, Theme } from '@material-ui/core';
import { useAuth0 } from "@auth0/auth0-react";
import { Inbox as InboxIcon, Person as PersonIcon } from '@material-ui/icons';
import SideNavMenuType from '../types/SideNavMenuType';

type Anchor = "left" | "right" | "top" | "bottom" | undefined

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    list: {
      background: "#154A40",
      color: "white",
      height: "100%",
    },
    header: {
      color: "white",
      fontWeight: 900,
      margin: "7px 20px",
      letterSpacing: "2px"
    },
    listItemIcons: {
      color: "white"
    },
  }),
);


const SideNavMenu = ({ sideMenu, toggleDrawer, handleOpenModal }: SideNavMenuType) => {
  const anchor: Anchor = "left";
  const classes = useStyles();
  const { logout, isAuthenticated } = useAuth0();

  const list = (anchor: Anchor) => (
    <div
      id="list-wrapper"
      className={classes.list}
      role="presentation"
      onClick={toggleDrawer(false)}
      onKeyDown={toggleDrawer(false)}
    >
      <List>
        <Typography variant="h4" gutterBottom className={classes.header}>
          Can't Decide
        </Typography>
        <Divider />
        {isAuthenticated && (
          <ListItem id="load-button" button key="load" onClick={(e: React.MouseEvent<HTMLElement>) => handleOpenModal(e, "load")}>
            <ListItemIcon className={classes.listItemIcons}><InboxIcon /></ListItemIcon>
            <ListItemText primary="Load" />
          </ListItem>
        )}
        {isAuthenticated && (
          <ListItem id="save-button" button key="save" onClick={(e: React.MouseEvent<HTMLElement>) => handleOpenModal(e, "save")}>
            <ListItemIcon className={classes.listItemIcons}><InboxIcon /></ListItemIcon>
            <ListItemText primary="Save" />
          </ListItem>
        )}
        <ListItem button key="logout" onClick={() => logout({ returnTo: window.location.origin })}>
          <ListItemIcon className={classes.listItemIcons}><PersonIcon /></ListItemIcon>
          <ListItemText className="logout" primary="Logout" />
        </ListItem>
      </List>
    </div>
  );

  return (
    <div>
      <React.Fragment key={anchor}>
        <Drawer anchor={anchor} open={sideMenu} onClose={toggleDrawer(false)}>
          {list(anchor)}
        </Drawer>
      </React.Fragment>
    </div>
  )
};

export default SideNavMenu;