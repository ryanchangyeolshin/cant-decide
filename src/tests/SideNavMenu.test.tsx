import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import SideNavMenu from '../components/SideNavMenu';
import SideNavMenuType from '../types/SideNavMenuType';

afterEach(cleanup);

describe("This suit is to test the SideNavMenu component", () => {
  const props: SideNavMenuType = {
    sideMenu: false,
    toggleDrawer: jest.fn((open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => "toggled"),
    handleOpenModal: jest.fn((event: React.MouseEvent<HTMLElement>) => "open modal!"),
  };

  test('Snapshot of SideNavMenu', () => {
    const { asFragment } = render(<SideNavMenu sideMenu={props.sideMenu} toggleDrawer={props.toggleDrawer} handleOpenModal={props.handleOpenModal} />);
    expect(asFragment()).toMatchSnapshot()
  });
});
