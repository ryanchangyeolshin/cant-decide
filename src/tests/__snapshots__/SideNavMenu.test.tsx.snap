// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the SideNavMenu component Snapshot of SideNavMenu 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
